demo-extended/build.gradle

```diff
@@ -30,4 +30,9 @@ dependencies {
     compile project(':library')
     compile 'com.android.support:appcompat-v7:23.1.0'
     compile 'com.android.support:design:23.1.0'
+
+    testCompile 'junit:junit:4.12'
+    testCompile 'org.mockito:mockito-core:1.10.19'
+    testCompile 'org.easytesting:fest-assert-core:2.0M10'
+    testCompile 'org.robolectric:robolectric:2.4'
```

Comment:
I think you can avoid using this if you mock both `UrlSpan` and `Patterns` to get the desired behaviour. For the second one you will have to make it a dependency of the class which is using it. 

Why not giving a try? I will take a look on my side as well to see if we can avoid using roboletric